src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs

```diff
@@ -135,7 +160,7 @@ private void ValidateState()
             {
                 case StreamState.Open:
                     return;
-                case StreamState.Disposed:
+                case StreamState.Closed:
                     throw new ObjectDisposedException(nameof(FrameRequestStream));
```

Comment:
1) To the user of the stream it should seem like its disposed; the implementation of it being reused shouldn't be exposed. Also in the non-reuse case it stays in the DIsposed state
2) Should throw as you are in the dangerzone writing to it in the reuse case.